packages/core/src/machines/actors/auth/signUp.ts

```diff
@@ -0,0 +1,229 @@
+
+import { AuthEvent, SignUpContext } from '../../../types';
+import { Auth } from 'aws-amplify';
+
+export const signUpActor = createMachine<SignUpContext, AuthEvent>(
+  {
+    id: 'signUpActor',
+    initial: 'init',
+    context: {
+      remoteError: '',
+      formValues: {},
+      validationError: {},
+    },
+    states: {
+      init: {
+        always: [
+          { target: 'confirmSignUp', cond: 'shouldInitConfirmSignUp' },
+          { target: 'signUp' },
+        ],
+      },
+      signUp: {
+        type: 'parallel',
+        exit: ['clearError', 'clearFormValues'],
+        states: {
+          validation: {
+            initial: 'pending',
+            states: {
+              pending: {
+                invoke: {
+                  src: 'validateFields',
+                  onDone: {
+                    target: 'valid',
+                    actions: 'clearValidationError',
+                  },
+                  onError: {
+                    target: 'invalid',
+                    actions: 'setFieldErrors',
+                  },
+                },
+              },
+              valid: { entry: sendUpdate() },
+              invalid: { entry: sendUpdate() },
+            },
+            on: {
+              CHANGE: {
+                actions: 'handleInput',
+                target: '.pending',
+              },
+            },
+          },
+          submission: {
+            initial: 'idle',
+            states: {
+              idle: {
+                entry: sendUpdate(),
+                on: {
+                  SUBMIT: 'validate',
+                  FEDERATED_SIGN_IN: 'federatedSignIn',
+                },
+              },
+              federatedSignIn: {
+                entry: [sendUpdate(), 'clearError'],
+                invoke: {
+                  src: 'federatedSignIn',
+                  onDone: '#signUpActor.resolved',
+                  onError: { actions: 'setRemoteError' },
+                },
+              },
+              validate: {
+                entry: sendUpdate(),
+                invoke: {
+                  src: 'validateFields',
+                  onDone: {
+                    target: 'pending',
+                    actions: 'clearValidationError',
+                  },
+                  onError: {
+                    target: 'idle',
+                    actions: 'setFieldErrors',
+                  },
+                },
+              },
+              pending: {
+                entry: [sendUpdate(), 'clearError'],
+                invoke: {
+                  src: 'signUp',
+                  onDone: {
+                    target: 'resolved',
+                    actions: ['setUser', 'setCredentials'],
+                  },
+                  onError: {
+                    target: 'idle',
+                    actions: 'setRemoteError',
+                  },
+                },
+              },
+              resolved: { type: 'final', always: '#signUpActor.confirmSignUp' },
```

Comment:
Note: I'm having explicit resolved state here so that `setCredentials` runs before `formValues` are cleared.